lib/startback/tracing/middleware.rb

```diff
@@ -0,0 +1,19 @@
+module Startback
+  module Tracing
+    class Middleware
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        context = ::Startback::Context::Middleware.context(env)
+        trace_id = env['HTTP_X_TRACE_ID']
+        parent_id = env['HTTP_X_SPAN_ID']
+        context.tracer = context.tracer.attach_to(trace_id, parent_id)
+        @app.call(env)
```

Comment:
1) Lets have unit tests for the middleware
2) Let make it robust to missing headers